design-system/packages/fields/src/DatePicker/index.tsx

```diff
@@ -31,6 +31,11 @@ export function useEventCallback<Func extends (...args: any) => any>(callback: F
   return cb as any;
 }
 
+function deserializeDate(value: string): Date {
+  const [year, month, day] = value.split('-').map(el => parseInt(el, 10));
+  return new Date(year, month - 1, day);
```

Comment:
Alternatively,  I think you could have

```suggestion
  return new Date(value.replace(/-/g, '/'))
```

Reply:
Actually, we should use `import { parse} from 'date-fns'`